pandas/core/internals/blocks.py

```diff
@@ -1989,6 +1987,12 @@ class DatetimeTZBlock(DatetimeLikeBlock):
     _validate_ndim = True
     _can_consolidate = False
 
+    def values_for_json(self) -> np.ndarray:
+        # force dt64tz to go through object dtype
+        # tz info will be lost when converting to
+        # dt64 which is naive
+        return self.values.astype(object)
```

Comment:
with this change, behavior is equivalent to just removing both this and DatimeLikeBlock.values_for_json.  Might mean a perf hit for dt64naive and td64 though

Reply:
Sorry, I'm not sure I follow here about dt64naive and td64. 
Don't those go through DatetimeLikeBlock instead of DatetimeTZBlock? (I didn't change any behavior there).

Reply:
right.  what im saying is that i think NDArrayBackedBlock.values_for_json works, so no longer needs to be overridden

Reply:
I'll take care of this in a follow up.